app/assets/javascripts/subscriptions.coffee

```diff
@@ -122,7 +122,17 @@ $(document).ready ->
           $('.estimate-text').text(data['estimate'])
           $('.granules-count-text').text(data['granules'])
           $('.frequency-text').text(data['frequency'])
-          $('.query-text').text("Query: #{$('#subscription_Query').val()}.")
+          query_text = $('#subscription_Query').val()
+          if (query_text.length > 300)  # more than 300 chars gets a text box
+            $('.query-text-large').text(query_text)
+            $('.query-text-small').hide()
+            $('.query-title').show()
+            $('.query-text-large').show()
+          else
+            $('.query-text-small').text("Query: #{query_text}")
+            $('.query-text-small').show()
+            $('.query-title').hide()
+            $('.query-text-large').hide()
```

Comment:
What happens when the user enters a valid query to get an estimation and then generates an error response (simple test case would be an invalid collection concept)?  I think the query text needs to be managed outside of the success/error check.

Reply:
Seems to be ok.   The error modal looks like this:
<img width="856" alt="Screen Shot 2020-05-28 at 2 29 48 PM" src="https://user-images.githubusercontent.com/54851481/83179363-efdbaa80-a0ef-11ea-9a22-a9795fd6b313.png">

Reply:
Apologies, I missed something.